Source/objects.cpp

```diff
@@ -1491,37 +1491,37 @@ void AddTorturedBody(Object &torturedBody)
 	torturedBody._oPreFlag = true;
 }
 
-void GetRndObjLoc(int randarea, int *xx, int *yy)
+Point GetRndObjLoc(int randarea)
 {
 	if (randarea == 0)
-		return;
+		return { 0, 0 };
 
 	int tries = 0;
+	int x;
+	int y;
 	while (true) {
 		tries++;
 		if (tries > 1000 && randarea > 1)
 			randarea--;
-		*xx = GenerateRnd(MAXDUNX);
-		*yy = GenerateRnd(MAXDUNY);
+		x = GenerateRnd(MAXDUNX);
+		y = GenerateRnd(MAXDUNY);
 		bool failed = false;
 		for (int i = 0; i < randarea && !failed; i++) {
 			for (int j = 0; j < randarea && !failed; j++) {
-				failed = !RndLocOk(i + *xx, j + *yy);
+				failed = !RndLocOk(i + x, j + y);
 			}
 		}
 		if (!failed)
 			break;
 	}
+	return { x, y };
 }
 
 void AddMushPatch()
 {
-	int y;
-	int x;
-
 	if (ActiveObjectCount < MAXOBJECTS) {
 		int i = AvailableObjects[0];
-		GetRndObjLoc(5, &x, &y);
+		auto [x, y] = GetRndObjLoc(5);

```

Comment:
We usually keep the use of auto to a minimum (lines where the type is already declared, such as casts).

```suggestion
		Point [x, y] = GetRndObjLoc(5);
```

Reply:
This requires C++17 and won't work on some of the compilers that we still support, such as GCC 6.

Reply:
Sorry, it escaped my mind that we support older compilers.

https://github.com/diasurgical/devilutionX/blob/master/docs/CONTRIBUTING.md might be a bit misleading then, saying that the oldest compiler used (GCC 6.5) supports most of C++17. Looks like most of the features are supported since GCC 7.

Reply:
Yeah that should be adjusted. It supports all of C++11 and some C++17 libraries via `std::experimental` (`string_view`, `optional`, etc).
C++17 language features, such as destructuring assignments, fold expressions, lambda auto arguments etc, are not supported by GCC 6.5.

Reply:
> Sorry, it escaped my mind that we support older compilers.

If you can allow CircleCI to run on your PR then it will check with the oldest relevant compiler so you don't have to think about it.